src/dvc_data/hashfile/meta.py

```diff
@@ -77,3 +77,14 @@ def to_dict(self) -> Dict[str, Any]:
 
 
 Meta.fields = list(fields_dict(Meta))
+
+META_SCHEMA = {
+    Meta.PARAM_ISDIR: bool,
+    Meta.PARAM_SIZE: int,
+    Meta.PARAM_NFILES: int,
+    Meta.PARAM_ISEXEC: bool,
+    Meta.PARAM_VERSION_ID: str,
+    Meta.PARAM_ETAG: str,
+    Meta.PARAM_CHECKSUM: str,
+    Meta.PARAM_MD5: str,
+}
```

Comment:
Not sure if this belongs here or in DVC `dvc.schema`

Reply:
Depends on where we save/load them.

Reply:
In the spirit of [Parse, don’t validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), may I suggest [`cattrs`](https://github.com/python-attrs/cattrs) here? It converts to appropriate type.

```pycon
>>> import cattrs
>>> cattrs.structure({'md5': 'b2c461b1b0bcb7a0a2318d5be1f13758.dir', 'size': 866190740, 'nfiles': '25002'}, Meta)
Meta(isdir=False, size=866190740, nfiles=25002, isexec=False, version_id=None, etag=None, checksum=None, md5='b2c461b1b0bcb7a0a2318d5be1f13758.dir')
```

Please also look at the benchmarks though, I don't really have an idea about performance here.